lib/bundles/inspec-compliance/api.rb

```diff
@@ -204,11 +204,22 @@ def self.is_compliance_server?(config)
     end
 
     def self.is_automate_server_pre_080?(config)
-      config['server_type'] == 'automate' && config['version'].nil?
+      # Automate versions before 0.8.x may have a "version" key in the config.
+      # Unless it's a hash that also contains a "version" key, it came from
+      # an Automate server that is pre-0.8.x.
+      return false unless config['server_type'] == 'automate'
+      return true unless config.key?('version')
+      return true unless config['version'].is_a?(Hash)
+      config['version']['version'].nil?
     end
 
     def self.is_automate_server_080_and_later?(config)
-      config['server_type'] == 'automate' && !config['version'].nil?
```

Comment:
Technically, a compliance server would test "true" for `!is_automate_server_pre_080?` too, right? I think that's another reason to leave this as proposed but I'd like to understand your position better.